proxy/create_container_interceptor.go

```diff
@@ -105,7 +110,17 @@ func (i *createContainerInterceptor) setWeaveWaitEntrypoint(container *docker.Co
 		}
 	}
 
+	for _, e := range container.Entrypoint {
+		if e != weaveWaitEntrypoint[0] {
+			return nil
+		}
+	}
```

Comment:
This looks just wrong to me. It will return, and hence not insert the `/w/w` when it finds something *other* than `/w/w`.